src/azure-cli-core/azure/cli/core/intercept_survey.py

```diff
@@ -0,0 +1,88 @@
+# --------------------------------------------------------------------------------------------
+
+import os
+import sys
+import json
+from datetime import datetime, timedelta
+from azure.cli.core._config import GLOBAL_CONFIG_DIR
+from azure.cli.core.style import print_styled_text
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+
+SURVEY_NOTE_NAME = 'az_survey.json'
+GLOBAL_SURVEY_NOTE_PATH = os.path.join(GLOBAL_CONFIG_DIR, SURVEY_NOTE_NAME)
+
+EXPERIENCE_PERIOD_IN_DAYS = 3
+PROMPT_INTERVAL_IN_DAYS = 180
+
+# VT code for text formatting:
+# https://docs.microsoft.com/en-us/windows/console/console-virtual-terminal-sequences#extended-colors
+SURVEY_STYLE = '\x1b[0;38;2;255;255;255;48;2;0;120;212m'  # Default & Foreground #FFFFFF & Background #0078D4
+
+# VT code for text modification:
+# https://docs.microsoft.com/en-us/windows/console/console-virtual-terminal-sequences#text-modification
+NEW_LINE = '\x1b[1L'
+ERASE_IN_LINE = '\x1b[0K'
+
+
+def should_prompt(cli):
+    if not cli.config.getboolean('core', 'survey_message', True):
+        return False
+
+    if sys.argv[1] == 'survey':
+        return False
+
+    if not os.path.isfile(GLOBAL_SURVEY_NOTE_PATH):
+        import uuid
+        from azure.cli.core._profile import Profile
+
+        # If the survey note file doesn't exist, then it should be the first time for cx to run CLI
+        # We should let cx try CLI for some days(EXPERIENCE_PERIOD_IN_DAYS) and then prompt the survey message
+        # We don't want to get the survey feedback on the same day, so we evenly distribute cx over 128 days
+        # using their installationId
+        installation_id = Profile(cli_ctx=cli).get_installation_id()
+        prompt_period = EXPERIENCE_PERIOD_IN_DAYS + (uuid.UUID(installation_id).int & 127)
+        next_prompt_time = datetime.utcnow() + timedelta(days=prompt_period)
+        survey_note = {
+            'last_prompt_time': '',
+            'next_prompt_time': next_prompt_time.strftime('%Y-%m-%dT%H:%M:%S')
+        }
+        with open(GLOBAL_SURVEY_NOTE_PATH, 'w') as f:
+            json.dump(survey_note, f)
+        return False
+
+    with open(GLOBAL_SURVEY_NOTE_PATH, 'r', encoding='utf-8-sig') as f:
+        survey_note = json.load(f)
+        next_prompt_time = datetime.strptime(survey_note['next_prompt_time'], '%Y-%m-%dT%H:%M:%S')
+        if datetime.utcnow() < next_prompt_time:
+            return False
+
+    return True
+
+
+def prompt_survey_message(cli):
+    if not should_prompt(cli):
+        return
+
+    # prompt message
+    print_styled_text((SURVEY_STYLE, NEW_LINE))
+    print_styled_text([
+        (SURVEY_STYLE, "[Survey] Help us improve Azure CLI by sharing your experience. "
+                       "This survey should take about 3 minutes. Run 'az survey' to open in browser. "
+                       "Learn more at https://go.microsoft.com/fwlink/?linkid=2203309"),
```

Comment:
The "Learn more" link leads to Azure CLI installation doc:

```
> curl https://go.microsoft.com/fwlink/?linkid=2203309 -i
HTTP/1.1 302 Moved Temporarily
Content-Length: 0
Server: Kestrel
Location: https://docs.microsoft.com/cli/azure/install-azure-cli
Request-Context: appId=cid-v1:26ef1154-5995-4d24-ad78-ef0b04f11587
X-Response-Cache-Status: True
Expires: Wed, 24 Aug 2022 08:35:50 GMT
Cache-Control: max-age=0, no-cache, no-store
Pragma: no-cache
Date: Wed, 24 Aug 2022 08:35:50 GMT
Connection: keep-alive
Strict-Transport-Security: max-age=31536000 ; includeSubDomains
```

This is a little bit confusing.

Reply:
The link will redirect to a new page once published. In the new page, we will share information about privacy and compliance, how to disable the survey message and more details about what we do with the survey data to users.